src/sentry/rules/conditions/event_frequency.py

```diff
@@ -339,18 +344,34 @@ def get_snuba_query_result(
         end: datetime,
         environment_id: int,
         referrer_suffix: str,
+        group_on_time: bool = True,
     ) -> Mapping[int, int]:
-        result: Mapping[int, int] = tsdb_function(
-            model=model,
-            keys=keys,
-            start=start,
-            end=end,
-            environment_id=environment_id,
-            use_cache=True,
-            jitter_value=group_id,
-            tenant_ids={"organization_id": organization_id},
-            referrer_suffix=referrer_suffix,
-        )
+        result: Mapping[int, int] = {}
+        if tsdb_function == tsdb.get_sums:
```

Comment:
I didn't want to add `group_on_time` to the other tsdb functions we may pass since they don't use it anyway, but this feels a little hacky.

Reply:
One option could be to use `functools.partial(tsdb.get_sums, group_on_time=True)` when you pass it, instead of passing it through here.

I think either that, or adding `group_on_time` as a param to all funcs is the right move